src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -429,6 +513,23 @@ private JsonRpcError ReadError(JToken json)
             };
         }
 
+        private void ClearProgressObject(object requestId)
```

Comment:
>object requestId [](start = 41, length = 16)

I don't see you doing any conversion within `ClearProgressObject` as you should to make sure that JToken, int, long or string (or whatever this number may appear as) gets converted properly.  What if you just take JToken here to make it clear that the caller needn't do any conversion and then use whatever method on JToken converts strings and integers to long.

Reply:
I change the param to receive a JToken object. `requestId.Value<long>()` will return the long value from an int or a string as long as the string represents an valid int number, otherwise it will throw a FormatException. Should I add a try-catch block for the casting? Or are we sure that we are always receiving a valid "long-castable" value there?

Reply:
That sounds great. It's OK to throw if the server sent us something we didn't expect (i.e. not an integer or string containing an integer). That will result in terminating the connection and logging out the error, which is appropriate.

---
In reply to: [303095032](https://github.com/milopezc/vs-streamjsonrpc/pull/1#discussion_r303095032) [](ancestors = 303095032)